packages/transformer/src/test/TestUtils/TimelineTestUtil.ts

```diff
@@ -525,9 +540,10 @@ export async function runTimeline(
 
         target.state = getIModelState(target.db); // update the tracking state
 
-        if (!isForwardSync)
+        if (!isForwardSync && !expectThrow)
```

Comment:
apparently I can't suggest through both since a line in the middle didn't change, but why not:

```ts
if (!expectThrow) {
  if (!isForwardSync)
    await saveAndPushChanges(accessToken, source.db, stateMsg);
  await saveAndPushChanges(accessToken, target.db, statMsg);
}
```

Reply:
changed